### ISSUE TITLE
Configure process will not complete with oneapi C compiler due to use of unsupported flag

### ISSUE THREAD

**Issue by frankwillmore** [HDFGroup/hdf5#750]
The configure process was observed to fail because of the unsupported warning flag -Wcheck:

```
CC=mpicc FC=mpifort MPICH_FC=ifx CXX=mpicxx ./configure --enable-parallel --enable-fortran
```

excerpt of config.log:
```
4480 configure:29054: checking how mpicc reports undeclared, standard C functions
4481 configure:29079: mpicc -c   -std=c99 -Wall -Wcheck     -D_GNU_SOURCE -D_POSIX_C_SOURCE=200809L      conftest.c >&5
4482 warning: unknown warning option '-Wcheck' [-Wunknown-warning-option]
4483 conftest.c:136:8: warning: implicitly declaring library function 'strchr' with type 'char *(const char *, int)' [-Wimplicit-     function-declaration]
4484 (void) strchr;
4485        ^
4486 conftest.c:136:8: note: include the header <string.h> or explicitly provide a declaration for 'strchr'
4487 2 warnings generated.
4488 configure:29079: $? = 0
4489 configure:29117: mpicc -c   -std=c99 -Wall -Wcheck     -D_GNU_SOURCE -D_POSIX_C_SOURCE=200809L      conftest.c >&5
4490 warning: unknown warning option '-Wcheck' [-Wunknown-warning-option]
4491 1 warning generated.
4492 configure:29117: $? = 0
4493 configure:29121: error: in `/home/willmore/hdf5-1.10.7':
4494 configure:29123: error: cannot detect from compiler exit status or warnings
CC=mpicc FC=mpifort MPICH_FC=ifx CXX=mpicxx ./configure --enable-parallel --enable-fortran
```

Removing the flag will allow the configuration, installation, and testing to complete, but does not address the underlying check in configure.

**Comment by frankwillmore** [comment]
see https://github.com/HDFGroup/hdf5/pull/751

**Comment by byrnHDF** [comment]
So we will need to do multiple things - remove the -Wcheck from general and put it in a "version 19 and under" file. Then add the appropriate checks and load in the intel-(cxx|f)flags file for autotools and the cmake/HDF5Compiler files.

**Comment by epourmal** [comment]
@frankwillmore we will be looking into the issues. Thank you for reporting!

**Comment by byrnHDF** [comment]
We have some fixes for this and some other flags issues in process.

**Comment by byrnHDF** [comment]
Develop updated with file changes for flags and will be merged down to other dev branches.

**Comment by frankwillmore** [comment]
@byrnHDF thanks for turning this around so quickly. Will this be available in a forthcoming release of 1.10? Asking because E4S is based on release versions.

**Comment by byrnHDF** [comment]
I believe 1.10.8 is scheduled for release soon after 1.12.1. (1.12.1 should be released within a week).
